daos-cortx/src/daos_kv_benchmark/kv_benchmarking.cc

```diff
@@ -0,0 +1,400 @@
+
+      buf = ( char * )calloc( KEY_DESC_BUF, sizeof( char ) );
+      d_iov_set( &sg_iov, buf, KEY_DESC_BUF );
+      sgl.sg_nr     = 1;
+      sgl.sg_nr_out = 0;
+      sgl.sg_iovs   = &sg_iov;
+
+      while ( !daos_anchor_is_eof( &anchor ) )
+      {
+         uint32_t nr = NR_QUERY;
+         int      rc;
+
+         memset( buf, 0, KEY_DESC_BUF );
+
+         rc = daos_kv_list( oh, DAOS_TX_NONE, &nr, kds, &sgl, &anchor, \
+                            NULL );
+         ASSERT( rc == 0, "KV list failed with %d", rc );
+
+         /* verify if returned number of descriptors are zero */
+         if ( nr == 0 )
+         {
+            continue;
+         }
+         else // if returned descriptors are non zero then compute each key and query value.
+         {
+            unsigned int offset = 0;
+
+            /* compute each key and fetch value */
+            for ( int i = 0; i < nr; i++ )
+            {
+               memset( key_buf, 0, key_size );
+               memset( rbuf, 0, val_size );
+
+               /* obtain key_buf value from sgl.sg_iovs */
+
+               printf( "key size is : %d\n", kds[ i ].kd_key_len );
+               memcpy( key_buf, ( char * )( ( sgl.sg_iovs )->iov_buf ) + offset, kds[ i ].kd_key_len );
+               printf( "key is : %s\n", key_buf );
+
+               /* update offset for next key */
+               offset += kds[ i ].kd_key_len;
+
+               /* compute value for the key_buf */
+               daos_size_t size = 0;
+               size = val_size;
+               daos_kv_get( oh, DAOS_TX_NONE, 0, key_buf, &size, rbuf, NULL );
+
+               printf( "rbuf is %s\n", rbuf );
+            }
+         }
+      }
+
+      free( ( char * )buf );
+   }
+
+   /* free resources */
+   free( ( char * )key_buf );
+   free( ( char * )val_buf );
+   free( ( char * )rbuf );
+
+   /* tear down */
+   tear_down( );
+}
+
+// bnechmark function is getting called here.
+static void kv_put_function( benchmark::State &state ) {
+   /* perform setup */
+   setup_main( );
+
+   unsigned int key_size = state.range( 0 );
+   unsigned int val_size = state.range( 1 );
+   unsigned int num_keys = state.range( 2 );
+
+   /* allocate key and value buffers */
+   char *key_buf = ( char * )calloc( key_size, sizeof( char ) ); // key buffer allocated
+   char *val_buf = ( char * )calloc( val_size, sizeof( char ) ); // value buffer allocated
+   memset( val_buf, 'z', val_size - 1 );                         // populate with some random value.
+
+   /* actual computation starts here */
+   for ( auto _ : state )
+   {
+      /* call daos_kv_put for state.range(2) times */
+      for ( int i = 0; i < num_keys; i++ )
+      {
+         state.PauseTiming( );
+
+         /* generate different key */
+         gen_key_names( key_buf, i, key_size );
+
+         state.ResumeTiming( );
+
+         /* actual function to mearsure time */
+         daos_kv_put( oh, DAOS_TX_NONE, 0, ( char * )key_buf, key_size, val_buf, NULL );
```

Comment:
replace key_size with size of value buffer

Reply:
done